### ISSUE TITLE
PrismaClient won't try to reconnect if database is down when the instance is created

### ISSUE THREAD

**Issue by doncicuto** [prisma/prisma#4782]
Hi, I am a noob with Prisma and so far I love it.

I'd like to point out that if I try the following code:

```
const express = require("express");
const { PrismaClient } = require("@prisma/client");
const prisma = new PrismaClient({ log: ["query"] });

const app = express();
app.get("/", async (req, res) => {
  try {
    const units = await prisma.units.findMany();
    res.json({ data: { units } });
  } catch (error) {
    res.status(500).json({ error: "Server error!" });
  }
});
app.listen(3100);
```

if my database is not running when I launch my Express server, my PrismaClient instance won't try to reconnect to the database if my database happens to listen for connections later (a recovery). PrismaClient will keep telling me that the server can't be reached.  

According to docs (https://www.prisma.io/docs/concepts/components/prisma-client/connection-management) a PrismaClient instance connects lazily when the first request is made, so I've added a check with prisma.$connect().then(() =>... in order to launch my Express app only if my database is ready from the very beginning. 

PrismaClient works fine if my server goes down and up again, and I know it's silly that you try to launch a server if your database is not there... but maybe other noobs like me would think that PrismaClient can try to reconnect with my database even if it was down when the server started. If you feel that this should be documented or added to Express examples, I can create a PR.

Thanks in advance

**Comment by pantharshit00** [comment]
Hey @doncicuto 

Generally you should use a script like `wait-for-it` to wait for a database connection: https://github.com/vishnubob/wait-for-it


However, if you really want retries in your code itself, you can roughly do it like so:
```js
const express = require("express");
const { PrismaClient } = require("@prisma/client");

let prisma;

const sleep = (amount = 0) =>
  new Promise((resolve) => setTimeout(resolve, amount));

const waitForDb = (maxAttempts = 5, attempt = 1) => {
  return new Promise(async (resolve, reject) => {
    console.log("Connection attempt to db", attempt, " max", maxAttempts);
    try {
      prisma = new PrismaClient();
      await prisma.$connect();
      return resolve();
    } catch (e) {
      if (attempt == 5) {
        return reject(new Error("Unable to connect to database"));
      } else {
        await sleep(5000);
        attempt++;
        return resolve(await waitForDb(maxAttempts, attempt));
      }
    }
  });
};

const app = express();

app.get("/", async (_, res) => {
  const users = await prisma.user.findMany();
  res.json(users);
});

waitForDb(5).then(() => {
  app.listen(5000, () => {
    console.log("app started on port 5000");
  });
});
```

**Comment by doncicuto** [comment]
Thanks @pantharshit00 for your sample code it will be helpful for those, like me, using Prisma with Express. Indeed I ended up checking if database was ready with prisma.$connect(). I created this issue after discussing it with @ryands17 at Slack channel so it may help others that would think that `new PrismaClient()` would check if database is ready and retry connection attempts.

Thanks again and happy new 2021!
